### DISCUSSION TITLE
Firefly III with HAProxy

### DISCUSSION THREAD

**Discussion by GFM-Network** [firefly-iii/firefly-iii#6801] [Firefly III]
Hello, im trying to run a self hosted firefly together with HAProxy but the problem is, all the pages of firefly has the addres:port/etc

Example:
1
![image](https://user-images.githubusercontent.com/72284266/210821722-866d8e6b-c344-4e64-b86a-e2f0d7d25db5.png)

2
![image](https://user-images.githubusercontent.com/72284266/210821874-5ea1d527-c4f9-4605-8846-8dc00f7463a5.png)

That's what happen with HAProxy:
![image](https://user-images.githubusercontent.com/72284266/210822008-b4155814-aaa2-4a51-9d06-d2e07db2bbbe.png)

the HAProxy backend is pointing to the right IP/PORT
![image](https://user-images.githubusercontent.com/72284266/210822166-305c05e9-66e0-4b8a-9539-2d383b4ca9dc.png)

If I remove the port from url the pages open but I can't use anything without the port
![image](https://user-images.githubusercontent.com/72284266/210822378-8679ed2f-20ec-42c8-9f8a-d092c39b19ef.png)

**Comment by JC5** [comment]
I'm not sure how HAProxy works. Can you edit the configuration of the reverse proxy somewhere?

**Comment by jstebenne** [comment]
What happens when you try to log in when accessing without the port number?

I am also using HAProxy and had an issue where the login form would submit to a `http` page instead of `https` and my browser would block it. This other discussion helped me: https://github.com/orgs/firefly-iii/discussions/6346#discussioncomment-3404435

**Reply by GFM-Network** [reply]
which section did you add into the backend?

**Reply by jstebenne** [reply]
Read the comment by **killtux**, in the page I linked:

> thanks for the suggestion. since i am using HAproxy, adding
>
> > http-request set-header X-Forwarded-Proto https
> 
> to the backend pool config worked :relaxed:
